### PR TITLE
feat: create profile list UI

### PR DESCRIPTION
**Pull request by hansac91**

### CODE REVIEW

**Comment by hansac91** [comment]
I need a tip how to solve img upload. Right now I just added it as a file to my branch and imported. Should I include it somehow in ProfileData.tsx? Or maybe create another file using Image component according to next.js documentation? if so, what should I include in src?

**Review by NerdbordMentor — APPROVED** [review]
## Code Review Report

### app/page.tsx
- No issues found. The code change adds the `ProfileList` component to the `Home` component.

### src/assets/images/ProfilePicture.png
- No issues found. The code change adds the `ProfilePicture.png` image.

### src/components/ProfileList/ProfileData.tsx
- No issues found. The code change adds an array of profile data.

### src/components/ProfileList/ProfileList.module.scss
- No issues found. The code change adds SCSS styles for the `ProfileList` component.

### src/components/ProfileList/ProfileList.tsx
- No issues found. The code change implements the `ProfileList` component according to the task requirements.
- The component renders a list of profile cards based on the `ProfileData` array.
- Each profile card displays the profile picture, name, seniority, location, and technologies.
- The `TechnologyChip` component is used to render the technology badges with appropriate styling based on the seniority and theme.
- The component handles different job specializations (Frontend, Backend, Fullstack) and assigns appropriate styles to the seniority and technology badges.
- The component also includes a title for the list of profiles.

Overall, the code changes seem to meet the requirements of creating a profiles list with engineer profile cards. The UI implementation closely matches the provided design, and the list handles both empty and non-empty states. Well done!

File: src/components/ProfileList/ProfileList.tsx [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+  name: string
+  theme: JobSpecialization
+  seniority: string
+}
+
+const getTechnologyClass = (theme: JobSpecialization, seniority: string) => {
+  if (seniority.includes('Frontend')) {
+    return styles.frontend
+  } else if (seniority.includes('Backend')) {
+    return styles.backend
```
Comment (Sumick) [inline src/components/ProfileList/ProfileList.tsx:36]: **nitpick:** We should split it into 2 separate functions, like: `getTechnologyClassByTheme`, and `getTechnologyClassBySeniority`, so they handle one argument

File: src/components/ProfileList/ProfileList.tsx [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+  }
+}
+
+const TechnologyChip: React.FC<TechnologyChipProps> = (props) => {
+  const themeClass = getTechnologyClass(props.theme, props.seniority)
+  return <p className={`${styles.badge} ${themeClass}`}>{props.name}</p>
+}
+
+const ProfileListItem: React.FC<ProfileListProps> = ({ data }) => {
+  let theme: JobSpecialization
```
Comment (Sumick) [inline src/components/ProfileList/ProfileList.tsx:59]: **nitpick:** We should use `useState` here for keeping, and updating such value: 
```
const [theme, setTheme] = useState(null)
```

File: src/components/ProfileList/ProfileList.tsx [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+  if (data.seniority.includes('Frontend')) {
+    theme = JobSpecialization.Frontend
+  } else if (data.seniority.includes('Backend')) {
+    theme = JobSpecialization.Backend
```
Comment (Sumick) [inline src/components/ProfileList/ProfileList.tsx:63]: **nitpick:** Then this should be set using `useEffect`, so it's:

```tsx
useEffect(() => {
 if (data.seniority.includes('Frontend')) {
    setTheme(JobSpecialization.Frontend)
 ...
}, [])
```

File: src/components/ProfileList/ProfileList.tsx [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+    </div>
+  )
+}
+
+const ProfileList: React.FC = () => {
+  return (
+    <div>
+      <div className={styles.title}>Profiles found</div>
+      <div className={styles.ProfileListCont}>
+        {ProfileData.map((profile) => (
```
Comment (Sumick) [inline src/components/ProfileList/ProfileList.tsx:114]: **nitpick:** variable names in javascript should be written in `camelCase`

**Review by Sumick — CHANGES_REQUESTED** [review]
Looks good - we need mainy to add React way of using logic in component. Using `useState`, and `useEffect` are the most important, and the most commonly used react hooks for handling logic.

**Review by wawrzonnn — APPROVED** [review]

**Review by Sumick — APPROVED** [review]
Looks nice, good job Hania!

**Pull request merged**